### DISCUSSION TITLE
SmoothStep op - Remove 0-1 limitation (Like in SmootherStep op)

### DISCUSSION THREAD

**Discussion by franckaubry** [cables-gl/cables_docs#215] [! Welcome - readme first!]
The SmoothStep and SmootherStep ops behave very differently, with the SmoothStep op being the more limited one.
Please remove the 0-1 limitation and add the possibility to SmoothStep between a custom range, even negative numbers like SmootherStep op can.

**Comment by pandrr** [comment]
both ops have the same min/max inputs do you mean another op ?

**Comment by franckaubry** [comment]
Well actually these two ops works differently when receiving the same exact numbers in any other range than 0 to 1.
If you feed them positive numbers (say 0 to 50), the SmootherStep op handles this as expected while the SmoothStep op caps out on 0 to 1.
Same when you feed them negative numbers. The SmoothStep op still stay between 0 to 1, while the SmootherStep op again behaves as expected.

Here is an example patch to describe the problem:
https://cables.gl/p/62500d584ca97bb437e2af02?s=CIUVoVy6IaozHbsW

**Comment by franckaubry** [comment]
Here is also a GIF to illustrate to problem further:

![Cables gl Smoother-Smooth bug-01](https://user-images.githubusercontent.com/97125854/165929032-856cad2c-fa19-4bb3-80f4-96eed7221caa.gif)

**Comment by pandrr** [comment]
thank you for the great example!

on dev.cables.gl there is a new version of smoothstep: SmoothStep_v2

with that op, the motion now behaves the same!

**Reply by franckaubry** [reply]
Thanks for fixing this!
I made an update to the patch as well to reflect these changes, with a switch to see the old/new op:
https://dev.cables.gl/p/Gd2J80

**Reply by franckaubry** [reply]
@pandrr If it makes any sense to use this patch as an example patch for the SmoothStep, I can clean it up/modify it.
Just let me know :)
